From 3e2af1c2ba6450075f0315a5230015853b97f87d Mon Sep 17 00:00:00 2001
From: Baptiste Mille-Mathias <baptiste.millemathias@gmail.com>
Date: Mon, 15 Nov 2010 14:21:09 +0000
Subject: Fix build against GNOME 3 platform

https://bugzilla.gnome.org/show_bug.cgi?id=634907
---
diff --git a/configure.ac b/configure.ac
index cc55705..bc5443a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,7 +85,7 @@ GNOME_DOC_INIT
 
 GTK_REQUIRED=2.91.1
 GLIB_REQUIRED=2.25.9
-GNOME_DESKTOP_REQUIRED=2.90.4
+GNOME_DESKTOP_REQUIRED=2.91.2
 GDKPIXBUF_REQUIRED=2.4.0
 # Require GConf 2.31+ for GConf-GSettings-Backend
 GCONF_REQUIRED=2.31.1
diff --git a/src/eog-file-chooser.c b/src/eog-file-chooser.c
index 4317017..ec5617a 100644
--- a/src/eog-file-chooser.c
+++ b/src/eog-file-chooser.c
@@ -34,7 +34,7 @@
 #ifndef GNOME_DESKTOP_USE_UNSTABLE_API
 #define GNOME_DESKTOP_USE_UNSTABLE_API
 #endif
-#include <libgnomeui/gnome-desktop-thumbnail.h>
+#include <libgnome-desktop/gnome-desktop-thumbnail.h>
 
 static char *last_dir[] = { NULL, NULL, NULL, NULL };
 
diff --git a/src/eog-thumbnail.c b/src/eog-thumbnail.c
index 05c0a4e..299f119 100644
--- a/src/eog-thumbnail.c
+++ b/src/eog-thumbnail.c
@@ -31,7 +31,7 @@
 #ifndef GNOME_DESKTOP_USE_UNSTABLE_API
 #define GNOME_DESKTOP_USE_UNSTABLE_API
 #endif
-#include <libgnomeui/gnome-desktop-thumbnail.h>
+#include <libgnome-desktop/gnome-desktop-thumbnail.h>
 
 #include "eog-thumbnail.h"
 #include "eog-list-store.h"
--
cgit v0.8.3.1
