# Description: the lpi changes require a configure update
# UbuntuSpecific: launchpad
#
diff -Nur eog-2.19.1/configure eog-2.19.1.new/configure
--- eog-2.19.1/configure	2007-04-30 23:54:27.000000000 +0200
+++ eog-2.19.1.new/configure	2007-05-02 10:21:52.000000000 +0200
@@ -24369,6 +24369,7 @@
              libgnomeui-2.0 >= $LIBGNOMEUI_REQUIRED \
              libglade-2.0 >= $LIBGLADE_REQUIRED \
              gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED \
+	     launchpad-integration \
              gconf-2.0 >= $GCONF_REQUIRED \
              libart-2.0 >= $LIBART_REQUIRED \
              gnome-desktop-2.0 >= $GNOME_DESKTOP_REQUIRED \
