--- trunk/src/eog-list-store.c	2007/09/22 12:07:56	4114
+++ trunk/src/eog-list-store.c	2007/09/23 02:37:03	4115
@@ -396,6 +396,14 @@
 			uri = gnome_vfs_uri_new (info_uri);
 			eog_list_store_append_image_from_uri (store, uri);
 			gnome_vfs_uri_unref (uri);
+		} else {
+			mimetype = gnome_vfs_get_mime_type (info_uri);
+			if (eog_image_is_supported_mime_type (mimetype)) {
+				uri = gnome_vfs_uri_new (info_uri);
+				eog_list_store_append_image_from_uri (store, uri);
+				gnome_vfs_uri_unref (uri);
+			}
+			g_free (mimetype);
 		}
 		break;
 
