diff -Nur -x '*.orig' -x '*~' eog-2.29.3/configure eog-2.29.3.new/configure
--- eog-2.29.3/configure	2009-11-27 21:31:17.000000000 +0100
+++ eog-2.29.3.new/configure	2009-12-01 15:57:20.000000000 +0100
@@ -13486,6 +13486,7 @@
              gio-2.0 >= $GLIB_REQUIRED \
              gnome-desktop-2.0 >= $GNOME_DESKTOP_REQUIRED \
              gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED \
+	     launchpad-integration \
              gconf-2.0 >= $GCONF_REQUIRED \
              gtk+-unix-print-2.0 >= $GTK_PRINT_REQUIRED \
 	     gnome-icon-theme >= $GNOME_ICON_THEME_REQUIRED \
